katsdpimager/loader_katdal.py

```diff
@@ -131,6 +135,10 @@ def __init__(self, filename, options):
             raise ValueError('Subarray {} is out of range'.format(args.subarray))
         if args.spw < 0 or args.spw >= len(self._file.spectral_windows):
             raise ValueError('Spectral window {} is out of range'.format(args.spw))
+        # Frequency-dependent private attributes contain data only for
+        # start_channel:stop_channel. The public interface adjusts by
+        # _start_channel to present channels 0:stop_channel.
```

Comment:
Will the loader ever be asked for e.g. channel 0? Or is that prevented by `args.start_channel`? Still trying to wrap my head around the difference in channel count...

Reply:
The loader will only ask about channels in start_channel:stop_channel. The constructor arguments are a hint about the channels that the loader will be asked about in future.

Reply:
Right, that's what I suspected.